### PR TITLE
Add /oauth/ro error documentation

### PR DESCRIPTION
**Pull request by ziluvatar**
Segmented by grant type and http error. This is an image of the final result inside of auth0-docs.

- [x] Writing reviewed
- [x] Possible security information leaked reviewed.

Closes: https://github.com/auth0/auth0-server/issues/2264

screencapture-errors-oauth-ro-1

screencapture-errors-oauth-ro-2

### CODE REVIEW

File: templates/sdk_auth_api.ejs [hunk 1]
```diff
@@ -126,6 +126,131 @@ Content-Type: 'application/json'
...
+    Missing connection parameter in the request:
+    <pre>{"error":"invalid_request","error_description":"missing connection parameter"}</pre>
+  </li>
+  <li>
+    Wrong format for scopes. List of scopes separated by whitespace is the right format.
+    <pre>{"error":"invalid_request","error_description":"scope parameter must be a string"}</pre>
+  </li>
+  <li>
+    If the connection/provider does not implement username/password authentication
+    <pre>{"error":"invalid_request","error_description":"specified strategy does not support requested operation"}</pre>
```
Comment (hernanhht) [inline templates/sdk_auth_api.ejs]: Wouldn't be better to have a space character after the `,` so the JSON can be in more than one line?

File: templates/sdk_auth_api.ejs [hunk 2]
```diff
@@ -126,6 +126,131 @@ Content-Type: 'application/json'
 </button>
 
 <pre class='tryme-result'><code id='ro-result'>Press the Try me! button to see the output result.</code></pre>
+
+<h4>Error Code Reference</h4>
+
+<h5>For jwt-bearer grant type</h5>
```
Comment (Catografix) [inline templates/sdk_auth_api.ejs]: Grant type: jwt-bearer

File: templates/sdk_auth_api.ejs (hunk 1 above)
Comment (ziluvatar) [inline templates/sdk_auth_api.ejs]: I haven't tried it, good one. Let me try.

File: templates/sdk_auth_api.ejs [hunk 3]
```diff
@@ -126,6 +126,131 @@ Content-Type: 'application/json'
...
+  <li>
+    Missing Device. You need to provide a device name for the caller device (like a browser, app, etc).
+    <pre>{"error":"invalid_request","error_description":"missing device parameter"}</pre>
+  </li>
+  <li>
+    Missing id_token. For this grant type you need to provide a JWT id token.
+    <pre>{"error":"invalid_request","error_description":"missing id_token parameter"}</pre>
+  </li>
+  <li>
+    Several errors related with an invalid id_token or its user:
```
Comment (Catografix) [inline templates/sdk_auth_api.ejs]: Errors related to an invalid id_token or user:

File: templates/sdk_auth_api.ejs [hunk 4]
```diff
@@ -126,6 +126,131 @@ Content-Type: 'application/json'
...
+    <pre>{"error":"invalid_grant","error_description":"..."}</pre>
+  </li>
+</ul>
+
+<h5>For password grant type</h5>
```
Comment (Catografix) [inline templates/sdk_auth_api.ejs]: Grant type: password

File: templates/sdk_auth_api.ejs (hunk 1 above)
Comment (ziluvatar) [inline templates/sdk_auth_api.ejs]: Nope it keeps in one line.

File: templates/sdk_auth_api.ejs [hunk 5]
```diff
@@ -126,6 +126,131 @@ Content-Type: 'application/json'
...
+  <li>
+    Missing password parameter in the request:
+    <pre>{"error":"invalid_request","error_description":"missing password parameter"}</pre>
+  </li>
+  <li>
+    Missing connection parameter in the request:
+    <pre>{"error":"invalid_request","error_description":"missing connection parameter"}</pre>
+  </li>
+  <li>
+    Wrong format for scopes. List of scopes separated by whitespace is the right format.
```
Comment (Catografix) [inline templates/sdk_auth_api.ejs]: Incorrect scope formatting. Each scope must  be separated by whitespace.

File: templates/sdk_auth_api.ejs [hunk 6]
```diff
@@ -126,6 +126,131 @@ Content-Type: 'application/json'
...
+    <pre>{"error":"invalid_request","error_description":"scope parameter must be a string"}</pre>
+  </li>
+  <li>
+    If the connection/provider does not implement username/password authentication
```
Comment (Catografix) [inline templates/sdk_auth_api.ejs]: The connection/provider does not implement username/password authentication:

File: templates/sdk_auth_api.ejs [hunk 7]
```diff
@@ -126,6 +126,131 @@ Content-Type: 'application/json'
...
+
+<h6>HTTP 401</h6>
+
+<ul>
+  <li>
+    Wrong username or password (this can vary depending on the identity provider):
+    <pre>{"error":"invalid_user_password","error_description":"Wrong email or password."}</pre>
+  </li>
+  <li>
+    Some other site had a security breach and your password has been leaked
```
Comment (Catografix) [inline templates/sdk_auth_api.ejs]: Another site had a security breach and your password has been leaked:

File: templates/sdk_auth_api.ejs [hunk 8]
```diff
@@ -126,6 +126,131 @@ Content-Type: 'application/json'
...
+
+<ul>
+  <li>
+    Some anomaly detections will return:
+    <pre>{"error":"too_many_attempts","error_description":"..."}</pre>
+    <pre>{"error":"too_many_logins","error_description":"..."}</pre>
+  </li>
+</ul>
+
+<h5>Common for any grant type</h5>
```
Comment (Catografix) [inline templates/sdk_auth_api.ejs]: All grant types

File: templates/sdk_auth_api.ejs [hunk 9]
```diff
@@ -126,6 +126,131 @@ Content-Type: 'application/json'
...
+  <li>
+    Missing client_id parameter in the request:
+    <pre>{"error":"invalid_request","error_description":"missing client_id parameter"}</pre>
+  </li>
+  <li>
+    Provided connection was not found:
+    <pre>{"error":"invalid_request","error_description":"the connection was not found"}</pre>
+  </li>
+  <li>
+    Provided connection is disabled. Check the connection in the dashboard, you might have turned it off for the provided client_id.
```
Comment (Catografix) [inline templates/sdk_auth_api.ejs]: Provided connection is disabled. Check the connection in the dashboard, you may have turned it off for the provided client_id.

File: templates/sdk_auth_api.ejs [hunk 10]
```diff
@@ -126,6 +126,131 @@ Content-Type: 'application/json'
...
+    <pre>{"error":"invalid_request","error_description":"the connection was disabled"}</pre>
+  </li>
+  <li>
+    The connection is not configured with properly custom scripts:
```
Comment (Catografix) [inline templates/sdk_auth_api.ejs]: The connection is not properly configured with custom scripts:

File: templates/sdk_auth_api.ejs [hunk 11]
```diff
@@ -126,6 +126,131 @@ Content-Type: 'application/json'
...
+    <pre>{"error":"invalid_request","error_description":"The connection is not yet configured..."}</pre>
+  </li>
+  <li>
+    The connection does not belong to the tenant. Check your base url.
+    <pre>{"error":"invalid_request","error_description":"the connection was not found for tenant..."}</pre>
+  </li>
+  <li>
+    In case you are using rules, some field name contains dots:
```
Comment (Catografix) [inline templates/sdk_auth_api.ejs]: If you are using rules, some field name contains dots:

File: templates/sdk_auth_api.ejs [hunk 12]
```diff
@@ -126,6 +126,131 @@ Content-Type: 'application/json'
...
+
+<h6>HTTP 403</h6>
+
+<ul>
+  <li>
+    Provided client_id is not valid:
+    <pre>{"error":"unauthorized_client","error_description":"invalid client"}</pre>
+  </li>
+   <li>
+    Some validations in specific points raised an access issue
```
Comment (Catografix) [inline templates/sdk_auth_api.ejs]: Validation of specific points raised an access issue:

File: templates/sdk_auth_api.ejs [hunk 13]
```diff
@@ -374,6 +499,19 @@ Content-Type: 'application/json'
...
+
+<h4>Error Code Reference</h4>
+
+<ul>
+  <li>
+    Wrong phone number or verification code:
+    <pre>{"error":"invalid_user_password","error_description":"Wrong phone number or verification code."}</pre>
+  </li>
+  <li>
+    See more errors for password grant in <a href="#db-and-ad">Database &amp; Active Directory / LDAP Authentication</a> /oauth/ro endpoint
```
Comment (Catografix) [inline templates/sdk_auth_api.ejs]: For more errors for the password grant type, see the: <a href="#db-and-ad">Database &amp; Active Directory / LDAP Authentication</a> /oauth/ro endpoint.

**Pull request merged**